File: productofarr_exceptself/product_of_arr_exceptself.js
function prodct_of_arr(arr) {
    const res = new Array(arr.length)

    let prefix = 1
    for(let i = 0; i < arr.length; i++) {
        res[i] = prefix
        prefix *= arr[i]
    }

    let postfix = 1

    for(let i = arr.length - 1; i >= 0; i--) {
        res[i] *= postfix
        postfix *= arr[i]
    }

    return res
}

File: productofarr_exceptself/productofarr_exceptself.py
def productofarr(vals):
    res = [1] * len(vals)

    prefix = 1
    for i in range(len(vals)):
        res[i] = prefix
        prefix *= vals[i]

    postfix = 1
    for i in range(len(vals) -1, -1, -1):
        res[i] *= postfix
        postfix *= vals[i]

    return res
